### PR TITLE
Chat messages

### PR DESCRIPTION
**Pull request by WickiRiama**
This PR adds :
- the possibility to send a message to a channel
- the channel name is displayed when you are on the channel room
- if you have not joined the channel, you can't see the content or send messages

To test:
- Connect with your username and with test.
- With your user: Create a channel and go to it (click on its name). Write a message. It should appear.
- With test user: The channel shouldn't appear. Search it (search bar) ang go to it. You shouldn't see any message. Try to write a message.
- With your user: The message typed by test shouldn't appear.
- In prisma, UserChannelMap, add a record with test in your channel.
- Now test user should be able to see the channel and its content.
- Do the same with new channel.
- Messages from one channel shouldn't appear in the other channel.
- Type a lot of messages. The last message should appear just above the input field and the message field should not expand.

### CODE REVIEW

File: backend/src/chat/chat.controller.ts [hunk 1]
```diff
@@ -50,4 +59,11 @@ export class ChatController {
 	async leaveChannel(@Param('channelId') channelId: string, @Req() req) {
 		return await this.chatService.leaveChannel(+channelId, req.user.sub);
 	}
+
+	@UseGuards(AuthGuard)
+	@Post('channel/update/')
```
Comment (basile41) [inline backend/src/chat/chat.controller.ts:64]: This route is not explicit enough, and why it is a POST when it should be a GET ?
```suggestion
	@Get('messages/:channelId')
```

Comment (WickiRiama) [reply backend/src/chat/chat.controller.ts:64]: The goal is toi have the channel name in the body of a POST request, to be able to check it with a DTO.

**Comment by SamanthaLHC** [comment]
- this is working in firefox and google chrome, all the messages are correctly sent, the display of multiple messages is good and scrolling is working.
- we can correctly navigate between differents channels with all the messages correctly displayed.

**Review by basile41 — APPROVED** [review]
Changed POST to GET request.
Changed route of GET request.
Removed 401 error when user hasn't joined the channel
Simplified isUserInChannel function

**Pull request merged**